sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -57,29 +57,48 @@ class CloudEvent(EventMixin):   #pylint:disable=too-many-instance-attributes
-    :param type: Required. Type of event related to the originating occurrence.
+    :keyword type: Required. Type of event related to the originating occurrence.
     :type type: str
-    :param time: The time (in UTC) the event was generated, in RFC3339 format.
+    :keyword time: The time (in UTC) the event was generated, in RFC3339 format.
     :type time: ~datetime.datetime
-    :param dataschema: Identifies the schema that data adheres to.
+    :keyword dataschema: Identifies the schema that data adheres to.
     :type dataschema: str
-    :param datacontenttype: Content type of data value.
+    :keyword datacontenttype: Content type of data value.
     :type datacontenttype: str
-    :param subject: This describes the subject of the event in the context of the event producer
+    :keyword subject: This describes the subject of the event in the context of the event producer
      (identified by source).
     :type subject: str
-    :param id: Optional. An identifier for the event. The combination of id and source must be
+    :keyword id: Optional. An identifier for the event. The combination of id and source must be
      unique for each distinct event. If not provided, a random UUID will be generated and used.
     :type id: Optional[str]
+    :ivar source: Required. Identifies the context in which an event happened. The combination of id and source must
+        be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
+    :vartype source: str
+    :ivar data: Event data specific to the event type.
+    :vartype data: object
+    :ivar type: Required. Type of event related to the originating occurrence.
+    :vartype type: str
+    :ivar time: The time (in UTC) the event was generated, in RFC3339 format.
+    :vartype time: ~datetime.datetime
+    :ivar dataschema: Identifies the schema that data adheres to.
+    :vartype dataschema: str
+    :ivar datacontenttype: Content type of data value.
+    :vartype datacontenttype: str
+    :ivar subject: This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :vartype subject: str
+    :ivar id: Optional. An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :vartype id: Optional[str]
     """
-    def __init__(self, source, type, **kwargs): # pylint: disable=redefined-builtin
-        # type: (str, str, Any) -> None
+    def __init__(self, source, type, data, **kwargs): # pylint: disable=redefined-builtin
+        # type: (str, str, object, Any) -> None
         self.source = source
         self.type = type
         self.specversion = kwargs.pop("specversion", "1.0")
         self.id = kwargs.pop("id", str(uuid.uuid4()))
         self.time = kwargs.pop("time", dt.datetime.now(UTC()).isoformat())
-        self.data = kwargs.pop("data", None)
+        self.data = data or None
```

Comment:
dumb question:
why `data` is a required parameter now if user could pass `data=None`?

Reply:
It's silent - users should still explicitly pass data as none

Reply:
Changed it a bit